Source/Engine/Debug/DebugDraw.cpp

```diff
@@ -629,18 +630,24 @@ void DebugDraw::DrawLine(const Vector3& start, const Vector3& end, const Color&
 
 void DebugDraw::DrawLines(const Span<Vector3>& lines, const Matrix& transform, const Color& color, float duration, bool depthTest)
 {
-    ASSERT(lines.Length() % 2 == 0);
+    if (lines.Length() % 2 == 0)
```

Comment:
Shouldn't this be `if (lines.Length() % 2 != 0)` ?

Reply:
No, you want the remainder to be zero for the division, if there is a remainder then it means the num of elements are uneven. As described, if this is the case you can have a start value but no end which is not valid.

Reply:
Here you're checking if the line length is *invalid* and then throw an error accordingly.
Basically `if(odd) { throw }`. Which means that the remainder should not be zero...

Reply:
> if there is a remainder then it means the num of elements are uneven
Right, but you're checking if there isn't a remainder, so if the array size is correct it'll throw

Reply:
Oops I read too quickly!